Lookup of dynamically created managers crashes on cached runs

# Bug report

Dynamically created manager nodes (built via `.from_queryset`) aren't cached. Thus any cached `"cross_ref"`s that points to such managers cannot be resolved by mypy. One example of a node having `"cross_ref"` set to this is `Model.objects` (which isn't a plugin generated node, so I think we need to cache the manager nodes)

Relates to `no_serialize=True` changes introduced in #828

## What's wrong

```
Traceback (most recent call last):
  File "/app/.venv/bin/mypy", line 8, in <module>
    sys.exit(console_entry())
  File "/app/.venv/lib/python3.10/site-packages/mypy/__main__.py", line 12, in console_entry
    main(None, sys.stdout, sys.stderr)
  File "mypy/main.py", line 96, in main
  File "mypy/main.py", line 173, in run_build
  File "mypy/build.py", line 180, in build
  File "mypy/build.py", line 256, in _build
  File "mypy/build.py", line 2717, in dispatch
  File "mypy/build.py", line 3041, in process_graph
  File "mypy/build.py", line 3119, in process_fresh_modules
  File "mypy/build.py", line 2008, in fix_cross_refs
  File "mypy/fixup.py", line 25, in fixup_module
  File "mypy/fixup.py", line 90, in visit_symbol_table
  File "mypy/fixup.py", line 45, in visit_type_info
  File "mypy/fixup.py", line 92, in visit_symbol_table
  File "mypy/nodes.py", line 932, in accept
  File "mypy/fixup.py", line 138, in visit_var
  File "mypy/types.py", line 970, in accept
  File "mypy/fixup.py", line 155, in visit_instance
  File "mypy/fixup.py", line 272, in lookup_fully_qualified_typeinfo
  File "mypy/lookup.py", line 47, in lookup_fully_qualified
AssertionError: Cannot find component 'MyManager' for 'myapp.models.MyManager'
make: *** [mypy] Error 1
```

## How is that should be

It shouldn't crash on cached runs and dynamically created manager nodes should be cached.

## System information

- OS:
- `python` version: 3.10.0
- `django` version: 4.0.1
- `mypy` version: 0.931
- `django-stubs` version: `master`
- `django-stubs-ext` version: 0.3.1

--- a/mypy_django_plugin/transformers/managers.py
+++ b/mypy_django_plugin/transformers/managers.py
@@ -217,7 +217,6 @@ def create_new_manager_class_from_from_queryset_method(ctx: DynamicClassDefConte
                 new_manager_info,
                 name=name,
                 sym_type=AnyType(TypeOfAny.special_form),
-                no_serialize=True,
             )
 
     # we need to copy all methods in MRO before django.db.models.query.QuerySet
@@ -278,9 +277,7 @@ def create_new_manager_class_from_from_queryset_method(ctx: DynamicClassDefConte
             )
 
     # Insert the new manager (dynamic) class
-    assert semanal_api.add_symbol_table_node(
-        ctx.name, SymbolTableNode(GDEF, new_manager_info, plugin_generated=True, no_serialize=True)
-    )
+    assert semanal_api.add_symbol_table_node(ctx.name, SymbolTableNode(GDEF, new_manager_info, plugin_generated=True))
 
 
 def fail_if_manager_type_created_in_model_body(ctx: MethodContext) -> MypyType:

